core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -209,17 +248,33 @@ public void restoreFromBackupLocal() throws Exception {
   }
 
   private void startStopTest(AlluxioMasterProcess master) throws Exception {
+    startStopTest(master, true, true);
+  }
+
+  private void startStopTest(AlluxioMasterProcess master,
+      boolean expectWebServiceStarted, boolean expectMetricsSinkStarted) throws Exception {
+    waitForAllServingReady(master, 5000);
+    assertTrue(expectWebServiceStarted == master.isWebServing());
+    assertTrue(expectMetricsSinkStarted == master.isMetricSinkServing());
+    master.stop();
+    assertFalse(isBound(mRpcPort));
+    assertFalse(isBound(mWebPort));
+  }
+
+  void waitForAllServingReady(AlluxioMasterProcess master, int timeoutMs)
+      throws InterruptedException, TimeoutException {
+    // Wait for the specific service port can be connected.
     waitForServing(ServiceType.MASTER_RPC);
     waitForServing(ServiceType.MASTER_WEB);
     assertTrue(isBound(mRpcPort));
     assertTrue(isBound(mWebPort));
     boolean testMode = ServerConfiguration.getBoolean(PropertyKey.TEST_MODE);
     ServerConfiguration.set(PropertyKey.TEST_MODE, false);
-    master.waitForReady(5000);
+    // Wait for the grpc service ready within 5 second.
+    master.waitForGrpcServerReady(timeoutMs);
+    // Wait for the web server ready within 5 second.
```

Comment:
Once you rename the methods to be self-explanatory, the comments are no longer needed. This line can be removed.

Reply:
Done.